### PR TITLE
IOS-12165: fix crash upon launching EMM because we try to extract use…

### PR DESCRIPTION
**Pull request by thuyBox**
…rID from a string userInfo

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @thuyBox has signed the CLA. Thanks for the pull request!

**Comment by boxtcarpel** [comment]
+2 LGTM

💃

File: BoxContentSDK/BoxContentSDK/Clients/BOXContentClient.m [hunk 1]
```diff
@@ -237,10 +237,12 @@ - (void)didReceiveAccessTokenRefreshedNotification:(NSNotification *)notificatio
 
 - (void)didReceiveUserWasLoggedOutNotification:(NSNotification *)notification
 {
-    NSDictionary *userInfo = (NSDictionary *)notification.object;
-    NSString *userID = [userInfo objectForKey:BOXUserIDKey];
-    if ([userID isEqualToString:self.user.modelID]) {
-        [self logOut];
+    if ([notification.object isKindOfClass:[NSDictionary class]]) {
```
Comment (RicoYao) [inline BoxContentSDK/BoxContentSDK/Clients/BOXContentClient.m:240]: I think this was caused by commit dda2df7eff30 (part of the App Users work).

Can we just do a refactor and stop using the 'object' for this. It really should go into userInfo. I believe there's only one component that posts this notification in BOXAPIDataOperation. It'd be great if we could just clean it up.

Comment (boxtcarpel) [reply BoxContentSDK/BoxContentSDK/Clients/BOXContentClient.m:240]: Agreed 100%. We had that discussion, that is IOS-12164. Thuy did not want to do that for 3.8.X releases, so we'll do it in Master for Hamilton. I want to do this next week.

**Pull request merged**